File: fsharp/mail/send_input.fs
open System
open System.Net.Mail

[<EntryPoint>]
let main(args: string[]) =
    if args.Length <> 4 then
        failwith "引数が必要 <SMTPサーバー> <From> <To> <Subject>"

    try
        let smtp = new SmtpClient(args.[0])
        //標準入力を読み込み
        let body = Console.In.ReadToEnd()

        smtp.Send(args.[1], args.[2], args.[3], body)
    with
    | _ as ex -> failwith ex.Message

    0

File: scala/mail/send_input.scala

import scala.io.Source
import org.apache.commons.mail.SimpleEmail

if (args.length != 4) {
	println("引数： <SMTPサーバー> <From> <To> <Subject>")
	exit
}

new SimpleEmail {
	charset = "UTF-8"
	hostName = args(0)
	setFrom(args(1))
	addTo(args(2))
	setSubject(args(3))
	setMsg(Source.stdin.mkString.replaceAll("\t", "　　"))
}.send

